src/modules/endpoint/reducers/pushCategory.js

```diff
@@ -4,15 +4,27 @@ import type { CategoriesStateType } from '../../app/StateType'
 import type { PushCategoryActionType } from '../../app/StoreActionType'
 import { CategoryModel } from '@integreat-app/integreat-api-client'
 
+/**
+ * Iterates through a tree until depth is reached. A depth of 0 means only the root is visited.
+ * A depth of 1 means the root and the children of it are visited. Please note that for each children of the root
+ * nodeAction is called once with the category and null for the children as parameters.
```

Comment:
Please note that for each node with depth `depth` null is passed as argument for `children` in nodeAction.